ledger/ledger-on-memory/src/app/scala/com/daml/ledger/on/memory/Main.scala

```diff
@@ -4,20 +4,34 @@
 package com.daml.ledger.on.memory
 
 import akka.stream.Materializer
+import com.daml.ledger.on.memory.InMemoryLedgerReaderWriter.Index
 import com.daml.ledger.participant.state.kvutils.app.LedgerFactory.SimpleLedgerFactory
 import com.daml.ledger.participant.state.kvutils.app.Runner
 import com.daml.ledger.participant.state.v1.{LedgerId, ParticipantId}
 import com.digitalasset.logging.LoggingContext
+import com.digitalasset.platform.akkastreams.dispatcher.Dispatcher
 import com.digitalasset.resources.{ProgramResource, ResourceOwner}
 
 import scala.concurrent.ExecutionContext
 
 object Main {
   def main(args: Array[String]): Unit = {
-    new ProgramResource(new Runner("In-Memory Ledger", InMemoryLedgerFactory).owner(args)).run()
+    val resource = for {
+      dispatcher <- InMemoryLedgerReaderWriter.dispatcher
```

Comment:
Seems like this line could be part of `InMemoryLedgerFactory`.

Reply:
I didn't know how to properly resolve the ownership of the dispatcher, because each call to `owner` (which will be made for each individual participant) should use the same dispatcher.

Reply:
Ah, if it needs to be shared, then as it is makes perfect sense.